### PR TITLE
Merge Makefiles

### PR DESCRIPTION
**Pull request by joshbwlng**
Reduce complexity and places in which environment variables
could be set or overridden.

Change-type: patch
Signed-off-by: Josh Bowling <josh@balena.io>

***

**Please remember to write tests for your changes**. We aim to automatically
deploy `master` to production, and we can't safely do this without a solid
test suite!

### CODE REVIEW

**Comment by joshbwlng** [comment]
@balena-ci rebase

File: Makefile [hunk 1]
```diff
@@ -240,7 +242,7 @@ else
 SCRUB_COMMAND =
 endif
 
-SENTRY_DSN_UI ?=
+SENTRY_DSN_UI ?= https://ff836b1e4abc4d0699bcaaf07ce4ea08@sentry.io/1366139
```
Comment (Hades32) [inline Makefile:245]: will this send my local issues to the prod sentry instance? Isn't this token secret?

**Review by Hades32 — COMMENTED** [review]
can't we get rid of all makefiles? I really don't understand why we would need them. Worst case we could put everything in npm run scripts...

File: Makefile (hunk 1 above)
Comment (joshbwlng) [reply Makefile:245]: @Hades32 We already had this in `apps/livechat/Makefile`, the idea was to move it here for simplicities sake. I wasn't sure if it was supposed to be a secret, but if so we probably need to regenerate it, it's been in clear text in this repo for ages 😓 

I get your point about it affecting local devenvs, we should explicitly set this to nothing for devenvs.

**Comment by joshbwlng** [comment]
| can't we get rid of all makefiles?

@Hades32 That's my plan, eventually get rid of all Makefiles (already did this for all JF lib repos if you haven't noticed). This one is a bit more difficult to drop, but I'm looking to slowly phase it out, especially after Livepush-based development is a bit less broken and people don't need to fallback to local development from time to time.

I guess we could just drop the Makefiles now and use a combination of npm scripts and possibly shell scripts if we need some extra logic for any reason. Might tackle that in an experimental branch next week or the next hack Friday. For now though, you can think of this PR as the first step in removing all Makefiles.

File: Makefile (hunk 1 above)
Comment (joshbwlng) [reply Makefile:245]: Oh, actually, we're not `export`ing it so I don't believe it's being passed to `dev-ui` or `dev-livechat`. It is being explicitly passed to `build-ui` and `build-livechat` which is to be expected.

**Comment by joshbwlng** [comment]
On second thought, instead of spending time and effort on a halfway solution, I'd rather drop Makefiles entirely once Livepush-based development is more stable. Will start working on dropping Makefiles in a separate PR next week.

**Pull request closed without merging**